src/apps/alpha/alpha.c

```diff
@@ -0,0 +1,128 @@
+        counter=get_char(c[i]);
+        SEGGER_RTT_printf(0, "Loading char: %c  (index %d)\n", c[i],counter);
+        if(counter!=-1){
+            load_output(m,letters[counter]);
+            vTaskDelay(pdMS_TO_TICKS(500));
+        }
+    }
+}
+
+void scroll_text(char* text){
+    Matrix_t *m = GetMatrix();
+    size_t len=strlen(text);
+    for(int i=0; i<len; i++){
+        if(text[i]==' '){
+            location=-1;
+            SEGGER_RTT_printf(0, "space\n");
+            for(int j=0; j<4;j++){
+                shift_matrix(m,0);
+                vTaskDelay(pdMS_TO_TICKS(100));
+            }
+            
+        }
+        else location=get_char(text[i]);
+        if(location!=-1){
+            for(int j=0; j<8; j++){
+                word[1][j] = letters[location][j];
+            }
+            for(int j=0; j<8; j++){
+                load_output(m,word[0]);
+                for(int k=0; k<8; k++){
+                    word[0][k] = (word[0][k] << 1);
+                    if((1<<7) & word[1][k]){
+                        word[0][k]++;
+                    }
+                    word[1][k] = (word[1][k] << 1);
+                }
+                shift_matrix(m,0);
+                vTaskDelay(pdMS_TO_TICKS(100));
+            }
+        }
+    }
+}
+
+void ButtonHandler(const Button *btn, ButtonEvent_t event)
+{
+    Matrix_t *matrix = GetMatrix();
+    if (btn == &sw2) {
+        switch(event) {
+            case BUTTON_EVENT_SHORT: 
+                SEGGER_RTT_WriteString(0, "SW2 Short Press\n"); 
+                if(counter<52) counter++;
+                else counter=0;
+                break;
+            case BUTTON_EVENT_LONG:  
+                SEGGER_RTT_WriteString(0, "SW2 Long Press\n"); 
+                counter=52;
+                break;
+            case BUTTON_EVENT_DOUBLE:
+                SEGGER_RTT_WriteString(0, "SW2 Double Tap\n"); 
+                if(counter>0) counter--;
+                else counter=52;
+                break;
+            default: break;
+        }
+    } else if (btn == &sw3) {
+        switch(event) {
+            case BUTTON_EVENT_SHORT: 
+                SEGGER_RTT_WriteString(0, "SW3 Short Press\n");             
+                if(counter>0) counter--;
+                else counter=52;
+                break;
+            case BUTTON_EVENT_LONG:  
+                SEGGER_RTT_WriteString(0, "SW3 Long Press\n"); 
+                counter=0;
+                break;
+            case BUTTON_EVENT_DOUBLE:
+                SEGGER_RTT_WriteString(0, "SW3 Double Tap\n"); 
+                if(counter<52) counter++;
+                else counter=0;
+                break;
+            default: break;
+        }
+    }
+    load_output(matrix,letters[counter]);
+}
+
+void vAlphaTask(void *pvParameters)
+{
+    scroll_text("Hola Mundo desde Alpha App");
+    
+    // After displaying, you can loop or delete the task
+    for(;;) {
+        vTaskDelay(pdMS_TO_TICKS(1000));
+    }
```

Comment:
Respecto a lo que hablamos del bucle infinito, creo que prefiero que eso lo maneje la app, es decir, en vez de tener una funcion que ponga la matriz en un scroll infinito, que la app se encargue periodicamente de scrollear el texto.
Fijate que si metes el scroll adentro del for infinito el scroll texto se repite con un 1s de demora
```suggestion
    
    // After displaying, you can loop or delete the task
    for(;;) {
        scroll_text("Hola Mundo desde Alpha App");
        vTaskDelay(pdMS_TO_TICKS(1000));
    }
```